app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.hedvig.android.core.designsystem.component.button.HedvigSecondaryContainedButton
+import com.hedvig.android.core.designsystem.component.information.HedvigInformationSection
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.icons.Hedvig
+import com.hedvig.android.core.icons.hedvig.normal.Info
+import com.hedvig.android.core.ui.clearFocusOnTap
+import com.hedvig.android.core.ui.dialog.ErrorDialog
+import com.hedvig.android.core.ui.infocard.VectorInfoCard
+import com.hedvig.android.core.ui.scaffold.HedvigScaffold
+import com.hedvig.android.data.travelcertificate.TravelCertificate
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryEvent
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryUiState
+import com.hedvig.android.feature.travelcertificate.CertificateHistoryViewModel
+import com.hedvig.android.feature.travelcertificate.data.TravelCertificateUri
+import hedvig.resources.R
+import kotlinx.datetime.LocalDate
+
+@Composable
+internal fun TravelCertificateHistoryDestination(
+  viewModel: CertificateHistoryViewModel,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  onShareTravelCertificate: (TravelCertificateUri) -> Unit,
+) {
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+  TravelCertificateHistoryScreen(
+    reload = { viewModel.emit(CertificateHistoryEvent.RetryLoadReferralData) },
+    onCertificateClick = { url ->
+      viewModel.emit(CertificateHistoryEvent.DownloadCertificate(url))
+    },
+    onContinue = onContinue,
+    navigateUp = navigateUp,
+    onShareTravelCertificate = onShareTravelCertificate,
+    uiState = uiState,
+  )
+}
+
+@Composable
+private fun TravelCertificateHistoryScreen(
+  reload: () -> Unit,
+  onCertificateClick: (String) -> Unit,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  onShareTravelCertificate: (TravelCertificateUri) -> Unit,
+  uiState: CertificateHistoryUiState,
+) {
+  var showBottomSheet by remember { mutableStateOf(false) }
+  val explanationSheetState = rememberModalBottomSheetState(true)
+  if (showBottomSheet) {
+    TravelCertificateInfoBottomSheet(
+      onDismiss = { showBottomSheet = false },
+      sheetState = explanationSheetState,
+    )
+  }
+
+  when (uiState) {
+    CertificateHistoryUiState.FailureDownloadingCertificate -> {
+      ErrorDialog(
```

Comment:
As discussed irl, let's turn the error scenario into something that shows inside the success screen.
That can then also not be a Dialog anymore, but a normal part of the screen.